Rocket.Chat/Models/Mapping/UserModelMapping.swift

```diff
@@ -36,5 +36,37 @@ extension User: ModelMappeable {
         if let utcOffset = values["utcOffset"].double {
             self.utcOffset = utcOffset
         }
+
+        if let emailsRaw = values["emails"].array {
+            let emails = emailsRaw.flatMap { emailRaw -> Email? in
+                let email = Email(value: [
+                    "email": emailRaw["address"].stringValue,
+                    "verified": emailRaw["verified"].boolValue
+                    ])
+
+                guard !email.email.isEmpty else { return nil }
+
+                return email
+            }
+
+            if let realm = realm {
+                let appendEmailsUsingRealmObjects = {
+                    emails.forEach({ email in
+                        let realmEmail = realm.create(Email.self, value: email, update: true)
+                        self.emails.append(realmEmail)
+                    })
+                }
+
+                if realm.isInWriteTransaction {
+                    appendEmailsUsingRealmObjects()
+                } else {
+                    try? realm.write {
+                        appendEmailsUsingRealmObjects()
+                    }
+                }
```

Comment:
Can't you use our extension of Realm here? Like this...

```swift
Realm.execute({ (realm) in
   ...
}
```

Reply:
It seems that I can't...  When I try `Realm.execute` it throws a "wrong thread exception" and `Realm.executeOnMainThread` it says that it is already in a write transaction. 

but, going after it I could found a way to improve this code, take a look:

```swift
if let realm = realm {
    emails.forEach({ email in
        if let realmEmail = realm.object(ofType: Email.self, forPrimaryKey: email.identifier as AnyObject) {
            self.emails.append(realmEmail)
        }
    })
} else {
    self.emails.append(contentsOf: emails)
}
```